radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -58,15 +56,57 @@ impl TransactionResult {
 
 #[derive(Debug, Clone, ScryptoSbor)]
 pub struct CommitResult {
+    pub state_updates: StateDiff,
+    pub state_update_summary: StateUpdateSummary,
     pub outcome: TransactionOutcome,
     pub fee_summary: FeeSummary,
-    pub actual_fee_payments: BTreeMap<ObjectId, Decimal>,
-    pub state_updates: StateDiff,
-    pub entity_changes: EntityChanges,
+    pub fee_payments: IndexMap<ObjectId, Decimal>,
     pub application_events: Vec<(EventTypeIdentifier, Vec<u8>)>,
     pub application_logs: Vec<(Level, String)>,
 }
 
+#[derive(Debug, Clone, ScryptoSbor)]
+pub struct StateUpdateSummary {
+    pub new_packages: Vec<PackageAddress>,
+    pub new_components: Vec<ComponentAddress>,
+    pub new_resources: Vec<ResourceAddress>,
+    pub balance_changes: IndexMap<Address, IndexMap<ResourceAddress, BalanceChange>>,
```

Comment:
As a future change, we could even just have the `object_parent_map` - and on the state_update, store the old values for update substates 👍 - we could calculate balance changes on demand upstream if we needed to - and save the work here.

Anyway, this is probably best left for another day.

Reply:
It's fairly easy to compute that `object_parent_map` - instead of passing `root` to recurisve traversal, we just need to pass the immediate parent (or even the full path).

My only worry is that we might be over-engineering stuff.

Reply:
Indeed.

In future if we have `object_parent_map` and old/new diffs, then that's enough to generate whatever we want from the receipt on demand later - eg these balance changes.

Personally I don't think that's over-engineering. It's the receipt giving us the minimal, most powerful information, which consumers can use to extract other insights.